azure-cli/utilities/deploy.utilities.ps1

```diff
@@ -29,4 +29,18 @@ function ConvertTo-Base64String {
 
   $bytes = [System.Text.Encoding]::UTF8.GetBytes($text)
   return [System.Convert]::ToBase64String($bytes)
+}
+
+function ConvertTo-RequestJson {
+  param (
+    [Parameter(ValueFromPipeline, Mandatory = $true)]
+    [object]
+    $Object,
+
+    [Parameter(Mandatory = $false)]
+    [int]
+    $Depth = 4
```

Comment:
Why would we want to set a depth that is less than maximum?

Reply:
I sat the depth to what I thought was the default depth for ConvertTo-Json as to not introduce another arbitrary limit that i myself have defined. However, it seems like the default value is 2. I will set update it to the same. 
User can always overwrite through the method or update the default value when copying the snippet code